node/toolrunner.ts

```diff
@@ -283,17 +319,90 @@ export class ToolRunner extends events.EventEmitter {
                     return Array.prototype.unshift.call(args, `"${arguments[0]}"`); // quote the file name
                 };
                 return args;
+            } else if (options.shell) {
+                let args: string[] = [];
+                for (let arg of this.args) {
+                    if (this._needQuotesForCmd(arg, '%')) {
+                        args.push(this._wrapArg(arg, '"'));
+                    } else {
+                        args.push(arg);
+                    }
+                }
+                return args;
             }
+        } else if (options.shell) {
+            return this.args.map(arg =>  {
+                if (this._isWrapped(arg, "'")) {
+                    return arg;
+                }
+                // remove wrapping double quotes to avoid escaping
+                arg = this._unwrapArg(arg.trim(), '"');
+                arg = this._escapeChar(arg, '"', "\\");
+                return this._wrapArg(arg, '"');
+            });
         }
 
         return this.args;
     }
 
+    /**
+     * Escape specified character.
+     * @param arg String to escape char in
+     * @param chars Char should be escaped
+     * @param escapePrefix Char which should be used to escape. "\" is used if not specified.
```

Comment:
This feels a little overparameterized - do we anticipate this ever using an escape prefix other than `\`? Parameterizing `charToEscape` seems more reasonable because I could see it being `"` or `'` very easily for example